tests/fail/both_borrows/retag_data_race_protected_read.rs

```diff
@@ -1,29 +0,0 @@
-//@revisions: stack tree
-//@compile-flags: -Zmiri-preemption-rate=0
-//@[tree]compile-flags: -Zmiri-tree-borrows
```

Comment:
We should have this as a `pass` test, with comments explaining which optimization is invalid because of this code being allowed.